tools/ci_build/github/android/build_aar_package.py

```diff
@@ -120,9 +120,13 @@ def _build_aar(args):
         # copy executables for each abi, in case we want to publish those as well
         abi_exe_dir = os.path.join(exe_dir, abi)
         for exe_name in ["libonnxruntime.so", "onnxruntime_perf_test", "onnx_test_runner"]:
+            src_exe_path = os.path.join(abi_build_dir, build_config, exe_name)
+            if not os.path.exists(src_exe_path):
+                continue
+
             os.makedirs(abi_exe_dir, exist_ok=True)
-            target_exe_name = os.path.join(abi_exe_dir, exe_name)
-            shutil.copyfile(os.path.join(abi_build_dir, build_config, exe_name), target_exe_name)
+            dest_exe_path = os.path.join(abi_exe_dir, exe_name)
```

Comment:
Is libonnxruntime.so required? I can understand the test exes being optional, but I would have expected we needed the onnxruntime library.

Reply:
it is required if there are any dynamically linked test exes. this is copying to `intermediates/executables/<abi>` which I believe is not used in the AAR.

the executables are published as a separate artifact:
https://github.com/microsoft/onnxruntime/blob/abdc31de401262bcb03f538423389c2eb264a0ce/tools/ci_build/github/azure-pipelines/templates/android-java-api-aar.yml#L78-L108
https://github.com/microsoft/onnxruntime/blob/abdc31de401262bcb03f538423389c2eb264a0ce/tools/ci_build/github/android/build_aar_and_copy_artifacts.sh#L39-L44